plugins/catalyst/lib/mqt-plugin.cpp

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2023 - 2025 Chair for Design Automation, TUM
+ * Copyright (c) 2025 Munich Quantum Software Company GmbH
+ * All rights reserved.
+ *
+ * SPDX-License-Identifier: MIT
+ *
+ * Licensed under the MIT License
+ */
+
+//===- mqt-plugin.cpp ------------------------------------*- C++ -*-===//
+//
+// This file is licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "mlir/Conversion/CatalystQuantumToMQTOpt/CatalystQuantumToMQTOpt.h"
+#include "mlir/Conversion/MQTOptToCatalystQuantum/MQTOptToCatalystQuantum.h"
+#include "mlir/Dialect/MQTOpt/IR/MQTOptDialect.h"
+#include "mlir/Dialect/MQTOpt/Transforms/Passes.h"
+
+#include <llvm/Config/llvm-config.h>
+#include <llvm/Support/Compiler.h>
+#include <mlir/IR/DialectRegistry.h>
+#include <mlir/Tools/Plugins/DialectPlugin.h>
+#include <mlir/Tools/Plugins/PassPlugin.h>
+
+using namespace mlir;
+
+/// Dialect plugin registration mechanism.
+/// Observe that it also allows registering passes.
+/// Necessary symbol to register the dialect plugin.
+extern "C" LLVM_ATTRIBUTE_WEAK DialectPluginLibraryInfo
+mlirGetDialectPluginInfo() {
+  return {MLIR_PLUGIN_API_VERSION, "MQTOpt", LLVM_VERSION_STRING,
+          [](DialectRegistry* registry) {
+            registry->insert<::mqt::ir::opt::MQTOptDialect>();
+          }};
+}
+
+/// The pass plugin registration mechanism.
+/// Necessary symbol to register the pass plugin.
+extern "C" LLVM_ATTRIBUTE_WEAK PassPluginLibraryInfo mlirGetPassPluginInfo() {
+  return {MLIR_PLUGIN_API_VERSION, "MQTOptPasses", LLVM_VERSION_STRING, []() {
+            mqt::ir::opt::registerMQTOptPasses();
```

Comment:
In the future, we will most likely want to expose all of our dialects and transformations/conversions here. Could be worth tracking that in an issue.